software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -20,49 +20,73 @@
 
 import java.util.Collection;
 
+import brooklyn.enricher.Enrichers;
 import brooklyn.entity.Entity;
 import brooklyn.entity.basic.AbstractEntity;
 import brooklyn.entity.basic.Entities;
 import brooklyn.entity.group.DynamicCluster;
 import brooklyn.entity.proxying.EntitySpec;
 import brooklyn.entity.trait.Startable;
+import brooklyn.event.AttributeSensor;
+import brooklyn.event.basic.Sensors;
 import brooklyn.location.Location;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
 public class RedisClusterImpl extends AbstractEntity implements RedisCluster {
 
-    protected RedisStore master;
-    protected DynamicCluster slaves;
+    private AttributeSensor<RedisStore> MASTER = Sensors.newSensor(RedisStore.class, "redis.master");
+    private AttributeSensor<DynamicCluster> SLAVES = Sensors.newSensor(DynamicCluster.class, "redis.slaves");
 
     public RedisClusterImpl() {
     }
 
     @Override
     public RedisStore getMaster() {
-        return master;
+        return getAttribute(MASTER);
     }
     
     @Override
     public DynamicCluster getSlaves() {
-        return slaves;
+        return getAttribute(SLAVES);
     }
     
     @Override
     public void start(Collection<? extends Location> locations) {
-        master = addChild(EntitySpec.create(RedisStore.class));
+        RedisStore master = getMaster();
+        if (master == null) {
+            master = addChild(EntitySpec.create(RedisStore.class));
+            setAttribute(MASTER, master);
+        }
+
         Entities.manage(master);
-        master.start(locations);
+        master.invoke(RedisStore.START, ImmutableMap.<String, Object>of("locations", ImmutableList.copyOf(locations))).getUnchecked();
 
-        slaves = addChild(EntitySpec.create(DynamicCluster.class)
-                .configure(DynamicCluster.MEMBER_SPEC, EntitySpec.create(RedisSlave.class).configure(RedisSlave.MASTER, master)));
-        slaves.start(locations);
+        DynamicCluster slaves = getSlaves();
+        if (slaves == null) {
+            slaves = addChild(EntitySpec.create(DynamicCluster.class)
+                    .configure(DynamicCluster.MEMBER_SPEC, EntitySpec.create(RedisSlave.class).configure(RedisSlave.MASTER, master)));
+            setAttribute(SLAVES, slaves);
+        }
+        
+        slaves.invoke(DynamicCluster.START, ImmutableMap.<String, Object>of("locations", ImmutableList.copyOf(locations))).getUnchecked();
 
         setAttribute(Startable.SERVICE_UP, calculateServiceUp());
+
+        addEnricher(Enrichers.builder()
+                .propagating(RedisStore.HOSTNAME, RedisStore.ADDRESS, RedisStore.SUBNET_HOSTNAME, RedisStore.SUBNET_ADDRESS, RedisStore.REDIS_PORT)
+                .from(master)
+                .build());
     }
 
     @Override
     public void stop() {
```

Comment:
See the pattern used in `DynamicCluster.stop()`, for setting stopping/stopped/on_fire.